src/commands/Quote.ts

```diff
@@ -0,0 +1,207 @@
+      this.findType = findType;
+      this.findTypeSize = findTypeSize;
+    }
+    execute(_client?: Client) {
+      if(this.discordMessage?.reference?.messageID != null){
+          this.discordMessage?.channel.messages.fetch(this.discordMessage?.reference?.messageID!)
+          .then(message => {
+            //Helpfull PreDebugging Tool
+            // console.log("findType")
+            // console.log(this.findType)
+            // console.log("findTypeSize")
+            // console.log(this.findTypeSize)
+            // console.log("generationType")
+            // console.log(this.generationType)
+            var value = -1
+            if (message.author.username == "TheLegend27"){
+              value = 0;
+            } else if (message.author.username == "Edison"){
+              value = 1;
+            } else if (message.author.username == "Satvik"){
+              value = 2;
+            } else if (message.author.username == "NKR"){
+              value = 3
+            } else if (message.author.username == "AweKingSome"){
+              value = 4
+            }
+            const data = fs.readFileSync('quotes.json', 'utf8');
+            const userArray = JSON.parse(data)
+            if(this.findTypeSize == 0){
+              try {
+                const quote = {
+                  "quote": message.content,
+                }
+                if (message.author.username == "TheLegend27") {
+                  userArray[0].push(quote)
+                } else if(message.author.username == "Edison") {
+                  userArray[1].push(quote)
+                } else if(message.author.username == "Satvik") {
+                  userArray[2].push(quote)
+                } else if(message.author.username == "NKR") {
+                  userArray[3].push(quote)
+                } else if(message.author.username == "AweKingSum") {
+                  userArray[4].push(quote)
+                }
+                fs.writeFileSync('quotes.json', JSON.stringify(userArray));
+                console.log("JSON data is saved.");
+              } catch (error) {
+                console.log("JSON data cannot be saved")
+                console.error(err);
+              }
+            }
+            else if(this.generationType == "find" && Object.keys(userArray[value]).length > 0 ){ 
+              try{
+                // :quote find #2 -> Still Works
+                if(this.findTypeSize == 2){
+                  let number = this.findType[1]
+                  if(this.findType[2] != null){
+                    number = number + this.findType[2]
+                  }
+                  if(this.findType[3] != null){
+                    number = number + this.findType[3]
+                  }
+                  if(this.findType[0] == "#"){
+                    if(parseInt(number) <= Object.keys(userArray[value]).length){
+                      this.discordMessage?.channel.send(userArray[value][parseInt(this.findType[1])-1].quote)
```

Comment:
Access the array using the parsed number, this only works with single digit numbers.